tools/testbench/testbench.c

```diff
@@ -588,8 +592,13 @@ static void test_pipeline_stats(struct testbench_prm *tp,
 	}
 	printf("Input sample (frame) count: %d (%d)\n", n_in, n_in / tp->channels_in);
 	printf("Output sample (frame) count: %d (%d)\n", n_out, n_out / tp->channels_out);
-	printf("Total execution time: %zu us, %.2f x realtime\n\n",
-	       delta, (double)((double)n_out / tp->channels_out / tp->fs_out) * 1000000 / delta);
+#if defined __XCC__
+	printf("Total execution cycles: %lu\n\n", (unsigned long)delta);
```

Comment:
```suggestion
	printf("Total execution cycles: " PRIu64 "\n\n", delta);
```
I know some people have a strong allergy to these macros so an alternative is:

```suggestion
	printf("Total execution cycles: %ju\n\n", (uintmax_t)delta);
```

Or maybe even simpler: if you're confident that `sizeof(unsigned long) >= sizeof(uint64_t)` in every toolchain, then why not just make `delta` an `unsigned long`?

Reply:
Wait, if you make `delta` a signed `long` then you can probably have even fewer casts.

Reply:
> if you're confident that `sizeof(unsigned long) >= sizeof(uint64_t)` in every toolchain, then why not just make `delta` an `unsigned long`?

@marc-hb I think in some cases `long` was still 32 bits and only `long long` was 64 bits

Reply:
Then make `delta` a `long long`, job done!

I never heard about LLP64 outside Windows:
https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models